### PR TITLE
Fixing the physics.units module and adding deprecation to useless functions

### LINKED ISSUE #22953: Deprecate "useless" methods in physics.units
There are a number of methods in `physics.units` that are marked as useless, e.g. 
https://github.com/sympy/sympy/blob/68022011872a20c27abd047125e5feb8ce3289cc/sympy/physics/units/dimensions.py#L573-L578

Some of these are deprecated, others are not. 

To be able to remove these at some stage, all methods that are mentioned as useless should be deprecated.

### PR DESCRIPTION
**Pull request by yasphy**
#### References to other Issues or PRs
Resolves #22953 

#### Brief description of what is fixed or changed
Adds Deprecation Warning to the useless functions in sympy.physics.units module

#### Release Notes

- physics.units
  - Deprecated Warning added to useless functions
  - ## BREAKING CHANGE
     - This PR depreciates `list_can_dims`,`dim`,`dim_vector`,`can_transf_matrix`,`inv_can_transf_matrix`,`dim_can_vector`,`print_dim_base` and `is_consistent` methods in physics.units module as mentioned in the aforementioned issue

### CODE REVIEW

**Comment by yasphy** [comment]
Can any of contributors tell me how to resolve this error?
image

**Comment by oscarbenjamin** [comment]
You need to set the `stacklevel` argument to `sympy_deprecation_wanring`.

**Comment by oscarbenjamin** [comment]
This is described in the deprecation policy:
https://docs.sympy.org/dev/guides/contributing/deprecations.html

**Comment by yasphy** [comment]
> You need to set the `stacklevel` argument to `sympy_deprecation_wanring`.

But the `stacklevel` argument takes `int` value. Which line should it point at?

**Comment by oscarbenjamin** [comment]
The `stacklevel` argument is supposed to be the number of frames between the caller and the place where the warning is issued. You can test this by running the tests locally.

**Comment by yasphy** [comment]
> The `stacklevel` argument is supposed to be the number of frames between the caller and the place where the warning is issued. You can test this by running the tests locally.

So can you tell which line should it point to as I am getting following error.
image

But I am pointing the `stacklevel` to the warning line
image

**Comment by oscarbenjamin** [comment]
Please don't paste screenshots unless absolutely necessary. It is much better to copy the text and place it in a code block.

The stacklevel is not the line number. It is the number of call frames between where the warning is raised and where it should be seen. It is probably supposed to be 1 or 2 or something.

**Comment by yasphy** [comment]
> Please don't paste screenshots unless absolutely necessary. It is much better to copy the text and place it in a code block.
> 
> The stacklevel is not the line number. It is the number of call frames between where the warning is raised and where it should be seen. It is probably supposed to be 1 or 2 or something.

Ok. Sorry for the screenshots

**Comment by yasphy** [comment]
> > Please don't paste screenshots unless absolutely necessary. It is much better to copy the text and place it in a code block.
> > The stacklevel is not the line number. It is the number of call frames between where the warning is raised and where it should be seen. It is probably supposed to be 1 or 2 or something.
> 
> Ok. Sorry for the screenshots

**Comment by yasphy** [comment]
> Please don't paste screenshots unless absolutely necessary. It is much better to copy the text and place it in a code block.
> 
> The stacklevel is not the line number. It is the number of call frames between where the warning is raised and where it should be seen. It is probably supposed to be 1 or 2 or something.

It isn't 1 or 2. How am I supposed to find the exact stacklevel? @oscarbenjamin

**Comment by oscarbenjamin** [comment]
I don't know the code in the units module and I haven't looked at it (and I'm not going to look at it) so I can't answer this question: you will have to investigate this yourself.

If you put an exception there instead of a warning then you can probably count the frames in the traceback.

**Comment by oscarbenjamin** [comment]
It looks like you're trying to edit the code through github rather than actually using git to clone the files down to your computer. If so then that's really not an efficient way to work on the PR.

**Comment by yasphy** [comment]
Ok

File: sympy/physics/units/tests/test_dimensionsystem.py [hunk 1]
```diff
@@ -18,8 +19,8 @@ def test_extend():
 
 def test_list_dims():
     dimsys = DimensionSystem((length, time, mass))
-
-    assert dimsys.list_can_dims == (length, mass, time)
+    with warns_deprecated_sympy():
```
[the remark this answers is not in the document]
Comment (yasphy) [reply sympy/physics/units/tests/test_dimensionsystem.py:22]: The issue was resolved.

**Comment by yasphy** [comment]
I suppose this resolves issue #22953. Also #22970 did help me for the warning part however, it was according to older deprecation policy.

**Comment by oscarbenjamin** [comment]
The commits in this PR will definitely need to be squashed but I suggest waiting for it to be reviewed first.

**Comment by yasphy** [comment]
So when would someone review it?

**Comment by yasphy** [comment]
I will create a final PR with all commits squashed.

**Pull request closed without merging**